action.yml

```diff
@@ -0,0 +1,43 @@
+name: Terraform Format
+description: Checks the terraform formatting, formats if broken and pushes the changes back to the branch. 
+runs:
+  using: "composite"
+  steps:
+    - name: Terraform Install
+      run: |
+        TERRAFORM_VERSION=`cat .terraform-version`
```

Comment:
Sometimes `.terraform-version` is not in the root folder of the repo but under subfolders.
Will this `cat` work properly in such cases?

Reply:
Good question, probably not! Do you have an example repo to test against?

That might also explain why the Jenkinsfile version of this set the version explicitly instead of reading the file.

Reply:
Looking at infrastructure-dev I'm not sure this applies. I have been testing this on package-iam (https://github.com/fac/package-iam/pull/56/checks), should have linked that in the PR.

I think the infra-* modules only contain .hcl files:

```
09Feb_15:14 markaddison@mark-addison-FA893:~/fac/infrastructure-live  (devp/aws-runner)
> find . -name '*.tf' | grep -v .terragrunt-cache | wc -l
       0
```

In infrastructure-modules I can find 396 .tf files but no .terraform-version! Does that sound right? I guess the version comes from the place the module gets pulled in? Currently the Jenkinsfile sets the version for fmt to 0.12.6.

I think as is this will work for single module package- repos.

I need to add an input to force the tf version when no .terraform-version is present, or add that file to repos. Maybe as a followup PR.

Reply:
Seems my find foo was rubbish. infrastructure-live has nested `.terrform-version` files (27 of them, they all say 0.12.29). Infrastructure-dev as well. Everything else basically has it at the root.

https://github.com/search?p=1&q=org%3Afac+filename%3A.terraform-version&type=Code